# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1328221950 0
#      Thu Feb 02 22:32:30 2012 +0000
# Branch gui
# Node ID df86157a49123fe23f21e494e1fb3489520a70d2
# Parent  3b6177bd63b052fc0f70d4511f546d5c6dd77dfc
Re-enable MSVC/Win32 compilation.

* gui/src/MainWindow.h: Include <QCloseEvent>.
* gui/src/msvc.pri: New file
* gui/src/src.pro: Use it.
* .hgsubstate: Update to current tip of qirc and qterminal repos.

diff --git a/.hgsubstate b/.hgsubstate
--- a/.hgsubstate
+++ b/.hgsubstate
@@ -1,3 +1,3 @@
 f9813bce2c06a6130a68db4478d1b16ddadaf276 gnulib
-86a267ce6ab8c3c9fa40f3e0dbc109a26479384e gui/qirc
-4535670b65621538da028b2e54b2ba30b7839f95 gui/qterminal
+34c2a274a1b607d2616d9903099a0905237c8f80 gui/qirc
+75e8afc065a08cbf8400dcb3386c5dc030cf8719 gui/qterminal
diff --git a/gui/src/MainWindow.h b/gui/src/MainWindow.h
--- a/gui/src/MainWindow.h
+++ b/gui/src/MainWindow.h
@@ -22,16 +22,17 @@
 #include <QThread>
 #include <QTabWidget>
 #include <QMdiArea>
 #include <QStatusBar>
 #include <QToolBar>
 #include <QQueue>
 #include <Qsci/qsciapis.h>
 #include <QMdiSubWindow>
+#include <QCloseEvent>
 #include "ResourceManager.h"
 #include "OctaveLink.h"
 #include "WorkspaceView.h"
 #include "HistoryDockWidget.h"
 #include "FilesDockWidget.h"
 #include "BrowserWidget.h"
 #include "lexer/lexeroctavegui.h"
 #include "QTerminal.h"
diff --git a/gui/src/msvc.pri b/gui/src/msvc.pri
new file mode 100644
--- /dev/null
+++ b/gui/src/msvc.pri
@@ -0,0 +1,11 @@
+CONFIG += release
+
+QMAKE_CXXFLAGS += -MP
+LIBDIR_SUFFIX = release
+
+msvc-debug {
+	QMAKE_CXXFLAGS_RELEASE += -Zi
+	QMAKE_CXXFLAGS_RELEASE -= -O2
+	QMAKE_LFLAGS_RELEASE += -debug
+	QMAKE_CLEAN += $(DESTDIR)$(QMAKE_TARGET).pdb
+}
diff --git a/gui/src/src.pro b/gui/src/src.pro
--- a/gui/src/src.pro
+++ b/gui/src/src.pro
@@ -22,19 +22,22 @@ TARGET              = octave-gui        
 DESTDIR             = ../bin                           # Destination of the output
 
 TRANSLATIONS        += languages/generic.ts \
                        languages/de-de.ts \
                        languages/pt-br.ts \
                        languages/es-es.ts \
                        languages/ru-ru.ts \
                        languages/uk-ua.ts           # Available translations
+
+win32-msvc*: include(msvc.pri)
+
 LIBS                += -lqscintilla2  \
-                       -L../qirc/libqirc -lqirc \
-                                           -L../qterminal/libqterminal -lqterminal \
+                       -L../qirc/libqirc/$$LIBDIR_SUFFIX -lqirc \
+                       -L../qterminal/libqterminal/$$LIBDIR_SUFFIX -lqterminal \
                         $$system(mkoctfile -p LIBS) \
                         $$system(mkoctfile -p OCTAVE_LIBS)
 
 mac {
     CONFIG -= app_bundle
 }
 
 # Includepaths and libraries to link against:
@@ -59,18 +62,17 @@ unix {
 }
 
 win32-g++ {
     QMAKE_LFLAGS += --enable-auto-import
 }
 
 win32-msvc* {
     DEFINES += QSCINTILLA_DLL
-    #CONFIG += console
-    include(msvc-debug.pri)
+    QMAKE_CXXFLAGS += -wd4244
 }
 
 QMAKE_LFLAGS        += $$LFLAGS $$system(mkoctfile -p RLD_FLAG)
 QMAKE_CXXFLAGS      += $$INCFLAGS
 
 # Files associated with the project:
 SOURCES +=\
     lexer/lexeroctavegui.cpp \
